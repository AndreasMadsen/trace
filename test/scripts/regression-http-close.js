'use strict';

const http = require('http');

//
// test an issue with http where events wheren't removed
//

http.createServer(function (req, res) {
  res.end('ho away :(');
}).listen(0, 'localhost', function listen() {
  const server = this;
  const addr = this.address();

  http.get('http://localhost:' + addr.port, function get(res) {
    res.on('data', function () {});
    res.once('end', function end() {

      server.close(function close() {
        throw new Error('trace');
      });
    });
  });
});
