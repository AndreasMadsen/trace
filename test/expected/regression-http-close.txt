/scripts/regression-http-close.js:20
        throw new Error('trace');
        ^

Error: trace
    at Server.close (/scripts/regression-http-close.js:r:c)
    at Object.onceWrapper (events.js:r:c)
    at Server.emit (events.js:r:c)
    at emitCloseNT (net.js:r:c)
    at process._tickCallback (internal/process/next_tick.js:r:c)
    at new TickObject (internal/process/next_tick.js:r:c)
    at nextTick (internal/process/next_tick.js:r:c)
    at Server._emitCloseIfDrained (net.js:r:c)
    at Server.close (net.js:r:c)
    at IncomingMessage.end (/scripts/regression-http-close.js:r:c)
    at Object.onceWrapper (events.js:r:c)
    at IncomingMessage.emit (events.js:r:c)
    at endReadableNT (_stream_readable.js:r:c)
    at process._tickCallback (internal/process/next_tick.js:r:c)
    at createServerHandle (net.js:r:c)
    at Server.setupListenHandle [as _listen2] (net.js:r:c)
    at listenInCluster (net.js:r:c)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:r:c)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:r:c)
    at Object.lookup (dns.js:r:c)
    at lookupAndListen (net.js:r:c)
    at Server.listen (net.js:r:c)
    at Object.<anonymous> (/scripts/regression-http-close.js:r:c)
    at Module._compile (internal/modules/cjs/loader.js:r:c)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:r:c)
    at Module.load (internal/modules/cjs/loader.js:r:c)
    at tryModuleLoad (internal/modules/cjs/loader.js:r:c)
    at Function.Module._load (internal/modules/cjs/loader.js:r:c)
    at Function.Module.runMain (internal/modules/cjs/loader.js:r:c)
    at startup (internal/bootstrap/node.js:r:c)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:r:c)
